### PR TITLE
Update core packages; fix referrer navigation; fix dist name

### PR DESCRIPTION
**Pull request by brandones**
This doesn't really fix referrer navigation all the way. It's actually quite broken, because the `useCurrentUser` hook is holding onto the cached user even after the user is logged out. There's probably a `refetchCurrentUser` needed somewhere, perhaps in the `CurrentUserContext`, but I don't know where. I still have no idea why esm-login has a second layer of user caching/management outside of esm-api's in the first place.

### CODE REVIEW

**Comment by FlorianRappl** [comment]
> I still have no idea why esm-login has a second layer of user caching/management outside of esm-api's in the first place.

Not sure here either, but IIRC the reason was that otherwise we've seen quite some unnecessary requests in there. With the caching (which was implemented *afterwards*) I am not sure if that's needed anymore. But the question here is: Does it really need to know this? It could just rely on its own information.

I think the flows here need to be redone and improved. The whole redirect logic is quite fragile.

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "@openmrs/esm-login-app",
   "version": "3.0.1",
   "description": "The login microfrontend for the OpenMRS SPA",
-  "main": "dist/openmrs-esm-login.js",
+  "main": "dist/openmrs-esm-login-app.js",
```
Comment (FlorianRappl) [inline package.json:5]: Why is this changed? I also don't see a change in the travis file.

If we change something as fundamental as this, then we need to make sure to be 100% consistent.

Comment (brandones) [reply package.json:5]: Makes the devtools port-to-URL interpolation work. Good catch about Travis! And yes, I agree we should be consistent about it.

**Comment by brandones** [comment]
> With the caching (which was implemented afterwards) I am not sure if that's needed anymore.

Ah, that makes sense, okay.

> Does it really need to know this?

Not sure I understand. The login app needs to check the user login status to determine whether to redirect to the target page.

> I think the flows here need to be redone and improved. The whole redirect logic is quite fragile.

I believe that. How would you recommend improving it? (unless you intend to fix it yourself at some point, in which case, please feel free :) )

**Comment by FlorianRappl** [comment]
I'll close this also as we now use GitHub actions and the given referrer fix has been applied already. Still, improving the login should be done at some point.
